src/java.base/share/classes/java/lang/foreign/MemorySegment.java

```diff
@@ -1312,6 +1312,44 @@ MemorySegment reinterpret(long newSize,
      */
     String getString(long offset, Charset charset);
 
+    /**
+     * Reads a string using the given byte length from this segment at the given offset,
+     * using the provided charset.
+     * <p>
+     * This method always replaces malformed-input and unmappable-character
+     * sequences with this charset's default replacement string. The {@link
+     * java.nio.charset.CharsetDecoder} class should be used when more control
+     * over the decoding process is required.
+     *
+     * @param offset  offset in bytes (relative to this segment address) at which this
+     *                access operation will occur
+     * @param length  byte length to be used for string conversion (not including any
```

Comment:
I suppose this might also create confusion - e.g. some users might expect the length to be "logical" (e.g. expressed in number of chars, which then can be turned into a physical length using the charset).